### DISCUSSION TITLE
.onChange

### DISCUSSION THREAD

**Discussion by karlmira** [danielsaidi/RichTextKit#192] [General]
Hi friends,

how can I use .onChange with RTE? When I add it to RichTextEditor(text: $brieftext, context: context) ... it is ignored.

**Comment by danielsaidi** [comment]
Do you mean `.onChange` on the text binding? The editor currently only changes the text every once in a while, since updating it too often will cause a re-render, but I'm thinking about adding a second binding, so that the original binding is continuously updates, while a second binding that is used by the editor only updates at certain occasions.

**Reply by karlmira** [reply]
Hi Daniel,

thanks for your reply. Yes, that's what I mean. I need it, to check, if there are some changes and display an alert before the window is closed unsaved.

**Comment by danielsaidi** [comment]
I should take some time and revisit this, then only use iOS 17 available APIs. That would let me use `@Observable` instead of the old APIs, which should improve performance quite a bit and make this workaround obsolete.

**Comment by S1D1T1** [comment]
see the discussion here: https://github.com/danielsaidi/RichTextKit/issues/64
I use .onChange to observe and change the text on each keystroke.
@Observable is a win, I suggest it if you don’t need to support pre ios17
